docs/proposals/proposal-trigger-and-deploy.md

```diff
@@ -0,0 +1,247 @@
+
+#### Project maintainer triggers pipeline to measure a release
+
+Calling the webhook will trigger the pipeline. The Report stage will provide
+the results to users of the project.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate?
+Think broadly.  For example, consider how this will impact or be impacted
+by scaling to support more CNCF Projects.
+
+How will this affect the benchmark tests, CNCF Project Maintainers, pipeline maintainers, etc?
+-->
+
+Multiple deployments will produce inaccurate results as we can only measure
+a single project per node. We can set concurrency in the workflow to ensure
+only a single instance runs at a time.
+
+Deployment may fail. What alerting do we need? Do we also need to notify the
+project?
+
+Uninstall at end of pipeline fails. We can wait till all flux finalizers are
+removed. In future we could create nodes on demand and delete on completion.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include manifests or workflow examples
+(though not always required) or even code snippets. If there's any ambiguity
+about HOW your proposal will be implemented, this is the place to discuss them.
+-->
+
+### Trigger
+
+CNCF projects will be given a GitHub webhook they can call to trigger the
+green reviews pipeline. The webhook will accept 3 inputs and trigger a GitHub
+Actions workflow.
+
+Inputs are
+
+- `cncf_project`: **required** Project to be deployed e.g. `falco`
+- `cncf_project_sub`: **optional** Subcomponent if project has multiple variants
+they wish to test e.g. `modern-ebpf`
+- `version`: **required** Version of project to be tested e.g. `0.37.0`
+
+```sh
+curl -X POST \
+     -H "Accept: application/vnd.github.v3+json" \
+     -H "Authorization: token $GITHUB_PAT" \
+     https://api.github.com/repos/cncf-tags/green-reviews-tooling/actions/workflows/pipeline.yaml/dispatches \
+     -d '{"ref":"main", "inputs": {"cncf_project": "falco", "cncf_project_sub": "modern-ebpf","version":"0.37.0"}}'
+```
+
+The CNCF projects will be given a GitHub fine grained access token limited to
+the `green-reviews-tooling` repo. This token will have
+
+- Read access to code and metadata
+- Read write access to actions
+
+### Deploy
+
+Flux will be used to deploy the CNCF project. Projects are able to use either
+`kustomization` or `helmrelease` resources to deploy their project.
+
+Project resources that should always be deployed in the cluster are stored in
+the current location in the tooling repo e.g. `/clusters/projects/falco/`
+and are reconciled by source-controller.
+
+When the pipeline executes it will look for yaml files in the projects dir.
+If there is a yaml file matching the `cncf_project` input its contents will be
+applied using kubectl. The same applies for the `cncf_project_sub` input. 
+
+The `version` param will need to be injected into the files to ensure the
```

Comment:
@dipankardas011 Did I understand correctly your idea is to use kustomize to inject the version param? Could you explain a bit more?

Reply:
I have added the commit hope it helps where we might move towards kustomize but still at initial phase lets make it very simple and use kubernetes raw manifests with repeations and later we can easily optimise, if we need!

Reply:
@dipankardas011 Good call, lets keep it simple for applying the manifests